components/config/useLocalPreferences.tsx

```diff
@@ -0,0 +1,90 @@
+import {
+  buildGraphicPreferenceKeyForLabel,
+  labelsRequiringMediaFilter,
+  GraphicMediaFilter,
+} from '@/common/labels'
+import { useLocalStorage } from 'react-use'
+
+export type GraphicMediaFilterPreference = Record<GraphicMediaFilter, boolean>
+
+// This is the container hook for interfacing on top of localstorage and storing device level preferences
+// There may be various types of preferences that can be stored here and ideally each preference category
+// will have its own hook that will utilize this
+export const useLocalPreferences = () => {
+  const initialValue: Record<string, Record<string, boolean>> = {
+    graphicMediaPrefs: {},
+  }
+
+  labelsRequiringMediaFilter.forEach((label) => {
+    ;(['blur', 'grayscale', 'translucent'] as GraphicMediaFilter[]).forEach(
+      (filter) => {
+        initialValue.graphicMediaPrefs[
+          buildGraphicPreferenceKeyForLabel(label, filter)
+        ] = true
+      },
+    )
+  })
+
+  const [localPreferences, setLocalPreferences] = useLocalStorage(
+    'ozoneLocalPreferences',
```

Comment:
Would there be any value in clearing these on logout, or keeping separate keys per user?

Reply:
I think I've avoided that until now since I see these configs as individual/device level preference, not account level. if the same user is accessing ozone using different accounts, I feel like they would want these configs across accounts as long as it's the same device. but yeah, I don't think the other approach to tie it to user level makes sense in some contexts.